testplan/runners/pools/base.py

```diff
@@ -259,25 +310,29 @@ def __init__(
         worker_type=Worker,
         worker_heartbeat=None,
         heartbeats_miss_limit=3,
-        task_retries_limit=3,
-        max_active_loop_sleep=5,
+        task_retries_limit=5,
         restart_count=3,
+        max_active_loop_sleep=5,
+        should_rerun=default_check_rerun,
         **options
     ):
         options.update(self.filter_locals(locals()))
         super(Pool, self).__init__(**options)
         self.unassigned = []  # unassigned tasks
         self.task_assign_cnt = {}  # uid: times_assigned
+        # Method `should_reschedule` should be implemented internally while
```

Comment:
Let's remove the should_reschedule code path for now, it doesn't add any value but could be quite confusing for ppl who is looking at this code. We can easily add it back if we need such in future.